tests/testthat/test-strings_as_factors_linter.R

```diff
@@ -20,58 +21,58 @@ test_that("strings_as_factors_linter skips allowed usages", {
 test_that("strings_as_factors_linter blocks simple disallowed usages", {
   expect_lint(
     "data.frame('a')",
-    rex::rex("This code relies on stringsAsFactors=TRUE"),
+    "This code relies on the default value of stringsAsFactors",
```

Comment:
We factor the message out to a variable in some cases. May be worth here too, to keep the diff small upon change in wording.

Reply:
i think there's a tradeoff with readability:

https://stackoverflow.com/questions/6453235/what-does-damp-not-dry-mean-when-talking-about-unit-tests

Reply:
Not sure I follow here, it would reduce the line count by half and highlight the relevant part, namely the input that should lint.
Readability is fine imo because the negative test cases are separate from the positive cases.

Reply:
Ah, I see. Keeping the variables local to the `test_that()` scope seems like a good trade-off to me. My IDE allows jumping to a definition and back quickly, so I favor less duplication.